brainiak/event_seg/event.py

```diff
@@ -0,0 +1,312 @@
+"""Event segmentation using a Hidden Markov Model"""
+
+# Authors: Chris Baldassano and Cătălin Iordan (Princeton University)
+
+import numpy as np
+from scipy import stats
+import logging
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "EventSegment",
+]
+
+
+class EventSegment:
+    """Class for learning and representing an event segmentation of
+    continuous fMRI data
+
+    Attributes
+    ----------
+    n_events: int
+        Number of segments to learn
+
+    step_var: lambda : default lambda step: 4 * (0.98 ** (step - 1))
+        The Gaussian variance to use during fitting, as a function of the
+        number of steps. Should decrease slowly over time.
```

Comment:
I was wrong about that. The type should be `Callable[[t1, t2, ..., tn], tr]`:
https://www.python.org/dev/peps/pep-0483/#fundamental-building-blocks

In case you are wondering, the requirement to document types like this in BrainIAK is documented (see second bullet):
https://github.com/IntelPNI/brainiak/blob/master/CONTRIBUTING.rst#standards